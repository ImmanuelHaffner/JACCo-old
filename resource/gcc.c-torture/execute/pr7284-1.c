





extern void abort (void);
extern void exit (int);

int
f (int n)
{
  return (n << 24) / (1 << 23);
}

volatile int x = 128;

int
main (void)
{
  if (f(x) != -256)
    abort ();
  exit (0);
}
