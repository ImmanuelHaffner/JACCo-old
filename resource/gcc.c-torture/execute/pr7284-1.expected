0
pr7284-1.c:7:1 keyword extern
pr7284-1.c:7:8 keyword void
pr7284-1.c:7:13 identifier abort
pr7284-1.c:7:19 punctuator (
pr7284-1.c:7:20 keyword void
pr7284-1.c:7:24 punctuator )
pr7284-1.c:7:25 punctuator ;
pr7284-1.c:8:1 keyword extern
pr7284-1.c:8:8 keyword void
pr7284-1.c:8:13 identifier exit
pr7284-1.c:8:18 punctuator (
pr7284-1.c:8:19 keyword int
pr7284-1.c:8:22 punctuator )
pr7284-1.c:8:23 punctuator ;
pr7284-1.c:10:1 keyword int
pr7284-1.c:11:1 identifier f
pr7284-1.c:11:3 punctuator (
pr7284-1.c:11:4 keyword int
pr7284-1.c:11:8 identifier n
pr7284-1.c:11:9 punctuator )
pr7284-1.c:12:1 punctuator {
pr7284-1.c:13:3 keyword return
pr7284-1.c:13:10 punctuator (
pr7284-1.c:13:11 identifier n
pr7284-1.c:13:13 punctuator <<
pr7284-1.c:13:16 constant 24
pr7284-1.c:13:18 punctuator )
pr7284-1.c:13:20 punctuator /
pr7284-1.c:13:22 punctuator (
pr7284-1.c:13:23 constant 1
pr7284-1.c:13:25 punctuator <<
pr7284-1.c:13:28 constant 23
pr7284-1.c:13:30 punctuator )
pr7284-1.c:13:31 punctuator ;
pr7284-1.c:14:1 punctuator }
pr7284-1.c:16:1 keyword volatile
pr7284-1.c:16:10 keyword int
pr7284-1.c:16:14 identifier x
pr7284-1.c:16:16 punctuator =
pr7284-1.c:16:18 constant 128
pr7284-1.c:16:21 punctuator ;
pr7284-1.c:18:1 keyword int
pr7284-1.c:19:1 identifier main
pr7284-1.c:19:6 punctuator (
pr7284-1.c:19:7 keyword void
pr7284-1.c:19:11 punctuator )
pr7284-1.c:20:1 punctuator {
pr7284-1.c:21:3 keyword if
pr7284-1.c:21:6 punctuator (
pr7284-1.c:21:7 identifier f
pr7284-1.c:21:8 punctuator (
pr7284-1.c:21:9 identifier x
pr7284-1.c:21:10 punctuator )
pr7284-1.c:21:12 punctuator !=
pr7284-1.c:21:15 punctuator -
pr7284-1.c:21:16 constant 256
pr7284-1.c:21:19 punctuator )
pr7284-1.c:22:5 identifier abort
pr7284-1.c:22:11 punctuator (
pr7284-1.c:22:12 punctuator )
pr7284-1.c:22:13 punctuator ;
pr7284-1.c:23:3 identifier exit
pr7284-1.c:23:8 punctuator (
pr7284-1.c:23:9 constant 0
pr7284-1.c:23:10 punctuator )
pr7284-1.c:23:11 punctuator ;
pr7284-1.c:24:1 punctuator }
